### PR TITLE
feat(split-panel): split panel support for ion-nav and ion-menu

### PR DESCRIPTION
**Pull request by manucorporat**
feb-16-2017 19-47-39

### CODE REVIEW

**Comment by intellix** [comment]
I know it's a WIP but just wondering if it'll support adding custom media queries instead of just small/medium/large?
`<ion-split-panel when="(min-width:600px)">`

**Comment by manucorporat** [comment]
@intellix I like the idea, probably something like:

**Comment by intellix** [comment]
SGTM, just cause it's in the previous one :D http://ionicframework.com/docs/api/directive/exposeAsideWhen/

**Comment by manucorporat** [comment]
@intellix it already does support custom media queries!

**Comment by numerized** [comment]
Hola, 

is there any way to use this features from the doc already to hide menu on login page for example or to show another menu for logged-in people ?

```
<ion-menu id="authenticated" side="left" [content]="mycontent">...</ion-menu>
<ion-menu id="unauthenticated" side="left" [content]="mycontent">...</ion-menu>
<ion-nav #mycontent [root]="rootPage"></ion-nav>
```

```
enableAuthenticatedMenu() {
  this.menuCtrl.enable(true, 'authenticated');
  this.menuCtrl.enable(false, 'unauthenticated');
}
```

File: src/components/split-panel/split-panel.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+ * @name SplitPanel
+ */
+@Component({
+  selector: 'ion-split-panel',
+  template: '<ng-content></ng-content>',
+  providers: [{provide: RootNode, useExisting: forwardRef(() => SplitPanel) }]
+})
+export class SplitPanel extends Ion implements RootNode {
+
+  _rmListerner: any;
```
Comment (intellix) [inline src/components/split-panel/split-panel.ts]: `_rmListener` :)

**Comment by highfieldjames** [comment]
Hi @manucorporat. Great work with this! I'm currently adding the split pane to multiple pages within my app.

Would you be willing to make ionChange an Output to follow the style of other components such as Checkbox?

**Comment by brandyscarney** [comment]
Found some small issues:

1. Content is not resizing in menu when the screen starts out small, see https://github.com/driftyco/ionic/issues/10518
2. Animation for push/pop seems off

**Comment by manucorporat** [comment]
@highfieldjames  what do you mean?

**Comment by highfieldjames** [comment]
@manucorporat What I mean is, at [line 86](https://github.com/driftyco/ionic/blob/7ec1cd021907731b17887207c5514bab09d337ce/src/components/split-panel/split-panel.ts#L86), shouldn't `ionChange: EventEmitter<SplitPanel> = new EventEmitter<SplitPanel>();` be `@Output ionChange: EventEmitter<SplitPanel> = new EventEmitter<SplitPanel>();` with `Output` added to the imports from `@angular/core`. It seems to be the case for other components such as [Checkbox](https://github.com/driftyco/ionic/blob/master/src/components/checkbox/checkbox.ts#L112) and [tabs](https://github.com/driftyco/ionic/blob/ac1a886158ddbb0211fc1e9e87c75b9d58e4664b/src/components/tabs/tabs.ts#L224).

**Comment by manucorporat** [comment]
@highfieldjames you are absolutely right!

**Comment by brandyscarney** [comment]
This looks great to me, I'm releasing a new nightly with it but I think it's good to merge.

**Comment by digaus** [comment]
Is it possible to define if the main pane is shown on the right or the left? So that when the screen is too small only the left pane will be shown? 

Currently I have two panes 
```
    <ion-nav [root]="rootPage" #content swipeBackEnabled="false" main></ion-nav>

    <ion-nav [root]="rootPage1" #content swipeBackEnabled="false"></ion-nav>
```
I want the main component to be shown on the left side rather then on the right to realise a detail page once the screen is big enought which will be shown on the right.

**Comment by seiyria** [comment]
You can change flex direction to be row reverse, that's how I solved it.

**Comment by digaus** [comment]
Could you post your code?

**Comment by seiyria** [comment]
Sure:

```
  .reversed-split-pane {
    flex-direction: row-reverse;

    ion-menu {
      border-right: none !important;
      // you'll also want to add border-left: 1px solid #dedede here
    }
  }
```

```
<ion-split-pane class="reversed-split-pane">
</ion-split-pane>
```

**Comment by manucorporat** [comment]
@seiyria @digaus 
https://github.com/driftyco/ionic/pull/10775

will be available in ionic 2.3

**Comment by seiyria** [comment]
Awesome, thank you!

**Comment by digaus** [comment]
Nice :)

**Comment by digaus** [comment]
Mhh not exactly what I was looking for... Here you can see what I want to do, currently switching rootPages to make it work which is not very good.
splitpane

**Comment by kodeine** [comment]
what if there are 2 menus (left/right). Using splitpane i cant close the right menu by default. looks like menuCtl functions dont work at all when using splitpane.

**Comment by kodeine** [comment]
no one?

**Comment by digaus** [comment]
You need to manually do that. Register a click event which triggers the right menu. Does not work otherwise

**Comment by kodeine** [comment]
@digaus this is the problem when using splitpane, none of the functions for menu work on browser. i cant close or open using menuCtrl.

**Comment by digaus** [comment]
@kodeine You also need to active the menu you want to open

**Comment by Farbdose** [comment]
When using ion-slides with horizontal scrolling b435ad6 causes a part of the next slide to be visible for +200ms when rotating from portrait to landscape as the ion-slides take that long to resize themselves.

_I don't have an absolute value but its long enough that after rotating the device you can see the left part of the next slide and wonder why it takes so long to resize while everything else is finished subjectively immediately_

@manucorporat Could you explain why this delay is needed to make ion-slides work correctly with split panes?
Also as a reminder in case you aren't aware anymore, you wrote:

> `// TODO: hacky, we should use Resize Observer in the future`

What prevented you from using Resize Observer in the first place and is the cause still valid?

**Pull request merged**